src/ui/public/config/config.js

```diff
@@ -129,34 +35,84 @@ module.service('config', function (Private, kbnVersion, kbnIndex, $rootScope, bu
    *                             be stored. Defaults to the config key
    * @return {function} - an unbind function
    */
-  config.$bind = function ($scope, key, property) {
-    if (!property) property = key;
-
-    let update = function () {
-      $scope[property] = config.get(key);
-    };
+  config.$bind = function (scope, key, property = key) {
+    config.on(key, update, scope);
+    function update(value) {
+      scope[property] = value;
+    }
+  };
 
+  config.on = function (key, fn, scope = $rootScope) {
+    const postfix = key === `*` ? `` : `.${key}`;
```

Comment:
Definitely something we could validate on. I think. But it'd mean custom keys would be required to be set *before* you can listen for changes on them. So maybe not a good idea. Maybe a warning? Even then seems weird in the case of custom values